### PR TITLE
Fixes #35759 - specify package evr when updating via errata to match upgrade-minimal

### PR DESCRIPTION
**Pull request by ianballou**
When installing an erratum via katello-agent on EL8 (with dnf), katello-agent tells libdnf to update the individual packages inside the erratum. Currently, it only passes the name, causing the latest RPM to be installed rather than the one in the erratum.  To fix this, I am appending the `evr` to the name to match the correct package.  The most ideal solution would be to use `dnf upgrade --minimal`, but `libdnf` doesn't seem to have Python bindings for that action.

To test:

1) `vagrant up rhel8` and register it to your Katello server.
2) Sync RHEL 8 AppStream + BaseOS.
3) Ensure that `tzdata-2022a-1.el8.noarch` and `tzdata-java-2022a-1.el8.noarch` are installed.
4) Locate the RHBA-2022:7067 and RHBA-2022:7404 errata in AppStream/BaseOS.  Note that the earlier one includes the `e` versions of the `tzdata` RPMs, and the latter one includes the `f` versions of the packages.
5) Install `katello-agent` on your RHEL 8 machine through the `satellite-client-6-for-rhel-8-x86_64-rpms` repo.
6) Apply the RHBA-2022:7067 erratum to your RHEL 8 machine and note that the wrong versions of the `tzdata` RPMs are installed: the `f` versions.
7) Rollback the `dnf update` from step (6).
8) Apply my patch to your RHEL 8 machine's katello-agent install. My related file was here:
```
/usr/lib/python3.6/site-packages/katello/agent/pulp/libdnf.py
```
9) Try the upgrade again and see that the proper `e` packages are installed.
10) Try this scenario on some other packages you can find with multiple errata.

### CODE REVIEW

**Review by sjha4 — COMMENTED** [review]
Seems like it was a conscious choice to drop evr from package based on this comment: https://github.com/Katello/katello-host-tools/blob/master/test/test_katello/test_agent/test_pulp/test_libdnf.py#L26
Remove that comment I guess.. 😄

**Comment by ianballou** [comment]
> Seems like it was a conscious choice to drop evr from package based on this comment: https://github.com/Katello/katello-host-tools/blob/master/test/test_katello/test_agent/test_pulp/test_libdnf.py#L26 Remove that comment I guess.. smile

@parthaa, this is a long shot, but do you remember why the `evr`s were taken out?

**Comment by ianballou** [comment]
The failing check relates to SUSE and doesn't seem to have anything to do with my PR.

**Comment by ehelms** [comment]
Is there an issue we can file somewhere to get upgrade minimal support added to libdnf to be more aligned to the intended use? Perhaps it's ugly, and a larger change, but have you considered changing this to use the CLI? Perhaps that is an enhancement on top of a short term fix. I would ask that we consider this though:

We want users to move away from katello-agent and to REX through SSH, Ansible or pull mode. We have different behaviors trying to hopefully achieve the same end user result:

 * katello-agent using update with constructed package name + evr
 * REX templates using update-minimal (https://github.com/Katello/katello/blob/master/app/views/foreman/job_templates/install_errata.erb#L22)
 * dnf / yum preferring the use of upgrade-minimal as the command 

If we end up with different behaviors through different methods we are potentially putting our users into an unknown situation and lack of trust of what is actually correct. 

All that is to say, the short term solution is a fine approach, I do think this raises some additional work on top to get some alignment here so we do not have different behaviors through different methods when to the user it's the same action -- apply errata.

**Review by sjha4 — APPROVED** [review]
Applying errata without change: 

Screenshot from 2022-11-15 16-20-15

Applying errata with change:

Screenshot from 2022-11-15 16-29-26

Test based ack 👍🏼 
Code LGTM. 👍🏼

**Comment by ianballou** [comment]
@ehelms I completely agree that applying advisories with a native libdnf API would be way better than our workaround. I made an issue: https://github.com/rpm-software-management/libdnf/issues/1586

**Comment by ianballou** [comment]
I'll need help merging this.

**Comment by sjha4** [comment]
I have merge access but I am waiting for one more ACK as discussed. 
Tagging smart people for the 2nd ack before we merge @parthaa @ehelms

**Comment by ianballou** [comment]
It looks like there may be a native way to install the errata. We need to test however that the upgrade is indeed minimal. Docs: https://dnf.readthedocs.io/en/latest/api_base.html#dnf.Base.add_security_filters

**Comment by ianballou** [comment]
I'm having a hard time getting the upgrade to work properly with the suggested errata-focused method, so for now let's focus on the fix as I have it here.  I asked some more questions on the libdnf issue.

**Review by parthaa — APPROVED** [review]
APJ

**Pull request merged**
